clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -850,6 +854,8 @@ public void subscribe(Collection<String> topics) {
     public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
         acquire();
         try {
+            if (pattern == null || pattern.pattern().equals("^\\s*$"))
```

Comment:
Hmm... Rejecting empty patterns may make sense, but it seems a little arbitrary. Why not reject patterns containing any whitespace since they couldn't match any topics? I think it would actually be OK to be more permissive and accept any regex since it is valid behavior for a pattern not to match any topics.